internal/pkg/aws/cloudwatch/cloudwatch.go

```diff
@@ -31,10 +32,19 @@ type resourceGetter interface {
 	GetResourcesByTags(resourceType string, tags map[string]string) ([]*rg.Resource, error)
 }
 
+type autoscalingAlarmNamesGetter interface {
+	ECSServiceAlarmNames(cluster, service string) ([]string, error)
+}
+
 // CloudWatch wraps an Amazon CloudWatch client.
 type CloudWatch struct {
-	cwClient api
-	rgClient resourceGetter
+	client api
+	// Optional client.
+	rgClient  resourceGetter
+	assClient autoscalingAlarmNamesGetter
+	// Optional client init.
+	initRgClient  func()
+	initAssclient func()
```

Comment:
OMG 😂🤣😂🤣😂🤣 I think you meant to type `AAS`

ok maybe we can actually get rid of `initAASClient` with the comment below and keep `rgClient` initialization the old way

Reply:
:hide: